compiler/rustc_mir_dataflow/src/impls/live_borrows.rs

```diff
@@ -0,0 +1,642 @@
+                    let node_idx =
+                        if let Some(node_idx) = self.locals_to_node_indexes.get(&borrowed_local) {
+                            *node_idx
+                        } else {
+                            self.dep_graph.add_node(NodeKind::Borrow(borrowed_local))
+                        };
+
+                    debug!(
+                        "adding edge from {:?}({:?}) -> {:?}({:?})",
+                        src_node_idx,
+                        self.dep_graph.node(src_node_idx).data,
+                        node_idx,
+                        borrowed_local
+                    );
+
+                    self.dep_graph.add_edge(src_node_idx, node_idx, ());
+                }
+                None => {}
+            },
+            _ => {}
+        }
+
+        self.super_place(place, context, location)
+    }
+
+    #[instrument(skip(self), level = "debug")]
+    fn visit_operand(&mut self, operand: &Operand<'tcx>, location: Location) {
+        self.super_operand(operand, location)
+    }
+
+    #[instrument(skip(self), level = "debug")]
+    fn visit_terminator(&mut self, terminator: &Terminator<'tcx>, location: Location) {
+        match &terminator.kind {
+            TerminatorKind::Call { destination, func, args, .. } => {
+                let dest_ty = destination.ty(self.local_decls, self.tcx).ty;
+                debug!(?dest_ty);
+
+                let mut has_refs_or_pointers_visitor = MaybeHasRefsOrPointersVisitor::new();
+                dest_ty.visit_with(&mut has_refs_or_pointers_visitor);
+
+                let dest_might_include_refs_or_pointers =
+                    if has_refs_or_pointers_visitor.has_refs_or_pointers {
+                        true
+                    } else if has_refs_or_pointers_visitor.has_params {
+                        // if we pass in any references or raw pointers as arguments and the
+                        // return type includes type parameters, these type parameters could
+                        // refer to those refs/ptrs, so we have to be conservative here and possibly
+                        // include an edge.
+                        true
+                    } else {
+                        false
+                    };
+
+                let node_idx =
+                    self.dep_graph.add_node(NodeKind::LocalOrLocalWithRefs(destination.local));
+                debug!("added local {:?} to graph with idx {:?}", destination.local, node_idx);
+                self.locals_to_node_indexes.insert(destination.local, node_idx);
+
+                if dest_might_include_refs_or_pointers {
+                    self.current_local = Some(node_idx);
+                    debug!("self.current_local: {:?}", self.current_local);
+                }
+
+                self.visit_operand(func, location);
+                for arg in args {
+                    self.visit_operand(arg, location);
+                }
```

Comment:
What if `arg` contains a pointer, which is stored somewhere, and returned later?
For instance:
```
enum Stash<T> { A, B(*const T) }

impl<T> Stash<T> {
    fn stash(&mut self, x: *const T) -> *const T {
        match *self {
            Stash::A => { *self = Stash::B(x); std::mem::null::<T>() }
            Stash::B(y) => { *self = Stash::B(y); x }
        }
    }
}

async fn foo<T>(x: T, y: T) {
    let mut stash = Stash::A;
    stash.stash(&raw const x);
    let p = stash.stash(&raw const y); // `p` points to `x`.
}
```